src/dune_async_io/async_io.ml

```diff
@@ -208,10 +209,34 @@ let rec select_loop t =
       Scheduler.fill_jobs fills);
     select_loop t
 
-let t_var = Fiber.Var.create ()
+let start t =
+  let module Scheduler = (val t.scheduler : Scheduler) in
+  Scheduler.spawn_thread (fun () ->
+      Mutex.lock t.mutex;
+      Exn.protect
+        ~f:(fun () -> select_loop t)
+        ~finally:(fun () -> Mutex.unlock t.mutex))
+
+module T_var : sig
```

Comment:
It would be nice if this module handled the start field exclusively. But we're still missing some handling in the finalizer. So my proposal would be either one of:

1. Remove the module completely
2. Move the conditional finalization to this module.

Reply:
I just did that. the goal was to make it impossible to get an unstarted `t` from the rest of the code.